ansible_base/authentication/utils/claims.py

```diff
@@ -352,58 +401,279 @@ def create_organizations_and_teams(results) -> None:
+
+    def _compute_system_permissions(self) -> None:
+        for role_name, has_permission in self.claims['rbac_roles'].get('system', {}).get('roles', {}).items():
+            if has_permission:
+                self.permissions_cache.add(role_name, organization=None, team=None)
+            else:
+                self.permissions_cache.remove(role_name, organization=None, team=None)
+
+    def _compute_organization_permissions(self) -> Iterator[Tuple[AbstractOrganization, dict]]:
+        orgs_by_name = self._get_orgs_by_name(self.claims['rbac_roles'].get('organizations', {}).keys())
+
+        for org_name, org_details in self.claims['rbac_roles'].get('organizations', {}).items():
+            if (org := orgs_by_name.get(org_name)) is None:
+                logger.info(_("Skipping organization '{organization}', because the organization does not exist").format(organization=org_name))
+                continue
 
-        org_names = claims['organization_membership'].keys() | claims['team_membership'].keys()
-        orgs_by_name = {org.name: org for org in Organization.objects.filter(name__in=org_names)}
+            for role_name, has_permission in org_details['roles'].items():
+                if has_permission:
+                    self.permissions_cache.add(role_name, organization=org)
+                else:
+                    self.permissions_cache.remove(role_name, organization=org)
+            yield org, org_details['teams']
 
-        for org_name, is_member in claims['organization_membership'].items():
-            org = orgs_by_name.get(org_name)
-            if org is None:
-                logger.info("Skipping organization '%s', because the organization does not exist", org_name)
-                continue
-            if is_member:
-                logger.info("Adding user '%s' to organization '%s'", user.username, org_name)
-                RoleDefinition.objects.managed.org_member.give_permission(user, org)
-            elif not remove_users:
-                logger.info("Removing '%s' from organization '%s'", user.username, org_name)
-                RoleDefinition.objects.managed.org_member.remove_permission(user, org)
-
-        for org_name, team_info in claims['team_membership'].items():
-            org = orgs_by_name.get(org_name)
-            if org is None:
-                logger.info("Skipping all teams in organization '%s', because the organization does not exist", org_name)
+    def _compute_team_permissions(self, org: AbstractOrganization, teams_dict: dict[str, dict]) -> None:
+        teams_by_name = self._get_teams_by_name(org.id, teams_dict.keys())
+
+        for team_name, team_details in teams_dict.items():
+            if (team := teams_by_name.get(team_name)) is None:
+                logger.info(
+                    _("Skipping team '{team}' in organization '{organization}', because the team does not exist").format(team=team_name, organization=org.name)
+                )
```

Comment:
We need to build the objects if they don't exist.

Reply:
we should rather log error. Teams and organizations are created during `create_claims()`, so this shouldn't happen

Reply:
Ah, ok. Yea, lets make this an error message and have it say something `because the team does not exist but it should already have been created`. That will also help clarify in the code that this shouldn't happen.

Reply:
messages updated